### ISSUE TITLE
Add SVG image support

### ISSUE THREAD

**Issue by Sata51** [jung-kurt/gofpdf#166]
Hi, It would be nice to support SVG image in PDF rendering. As input file as 

```
pdf.Image(p.FileName, leftPosition, topPosition, imageWidth, imageHeight, false, "", 0, "")
```

**Comment by jung-kurt** [comment]
I agree. Are you aware of any package that can read an SVG into a structure that satisfies the standard library [Image interface](https://golang.org/pkg/image/#Image)? if we find one, it would be straightforward to support this. However, in order to keep gofpdf dependent only on the standard library, such support would go into the contrib directory. All this means is that programs that require SVG support would need to import an additional package.

**Comment by joewestcott** [comment]
@jung-kurt Do you think it's better to find an external package to do the rasterisation for us, as described above, or to continue with our implementation of parsing and drawing from the SVG code?

The former sounds easier, but the rasterisation process involved will come with a trade off of quality vs file size for the resulting PDF. With the latter, we'd have to write/maintain a great deal of SVG parsing code. Interested to hear your thoughts on this.

**Comment by jung-kurt** [comment]
This is a really good question. It's a shame to rasterize an SVG image when PDF supports most of its underlying graphic commands. If you have a particular use case that isn't too complex, it might be worth the effort to extend the very basic support that gofdpf has, although I can't vouch for how robust its parser is. I wonder if @ajstarks (author of [svgo](https://github.com/ajstarks/svgo) has any remarks.

**Comment by ajstarks** [comment]
As you know svgo is a generator, not a rasterizer, however @srwiley has some interesting work on rasterizing SVG: see https://github.com/srwiley/oksvg and https://github.som/srwiley/scanx and rasterx

**Comment by jung-kurt** [comment]
Thanks, @ajstarks. Do you know if any work has been done to map SVG commands to PDF's vector commands? For example, gofpdf supports rudimentary, path-only SVGs by mapping M, C and L commands to "move to", Bézier curve and "line to" PDF primitives respectively. Simple line paths can thus be rendered without any rasterization. You probably know more than anyone how big a job it would be to extend this code to support a larger part of the SVG specification.

**Comment by ajstarks** [comment]
The SVG spec is quite large -- what subset would you want to support?

**Comment by jung-kurt** [comment]
For my purposes, it would be sufficient to support simple charts without text or embedded images. But @joewestcott, who rekindled this thread, may have other requirements.

**Comment by ajstarks** [comment]
Interesting you mention charts; note the dchart program, (https://github.com/ajstarks/deck/blob/master/cmd/dchart/README.md) which uses the gofpdf client, pdfdeck, for PDF rendering  (https://github.com/ajstarks/deck/blob/master/cmd/pdfdeck/pdfdeck.go)

**Comment by jung-kurt** [comment]
Perfect! This will be a great resource.
